### PR TITLE
Enable gpg signing for gitlab

### PR DESCRIPTION
**Pull request by zadjadr**
Allows to add a key to sign automated commits in gitlab according to https://docs.gitlab.com/ee/administration/gitaly/configure_gitaly.html#configure-commit-signing-for-gitlab-ui-commits

### CODE REVIEW

File: init.sls [hunk 1]
```diff
@@ -12,6 +12,19 @@ gitlab_repo:
 {{ edition }}:
   pkg.installed: []
 
+# Deploy signing_key.gpg if options are given in pillar.
+# Make sure it's not existent if not.
+{% if salt['pillar.get']('gitlab:gitaly:signing_key', None) != none %}
```
Comment (chr4) [inline init.sls:17]: As `/etc/gitaly` is not present in the default install, maybe we can place the keys within `/etc/gitlab`? 

Also, I think we should support SSH keys. We could also infer the filename from pillars, something like this

```yaml
gitaly:
  signing_key:
    gitaly_ui.ssh: 'ssh-ed25519 AAAA...'

    # Alternatively
    gitaly_ui.gpg: |
       ---- BEGIN -----
```

I'm not sure if the file extention is somehow valued, if it's inferred from the content we could just use `/etc/gitlab/gitaly-ui.key` instead.

Comment (zadjadr) [reply init.sls:17]: SSH keys should be supported with the proposed solution already, as you can only provide one signing_key and that signing_key will then just be used by gitaly. If you want to use a ssh key, you just have to provide it as `gitlab.gitaly.signing_key`

Comment (zadjadr) [reply init.sls:17]: Regarding changing the path: Sounds good :+1:

File: pillar.example [hunk 2]
```diff
@@ -33,6 +33,11 @@ gitlab:
       # Run at a specific time, defaults to 02:00 every night
       on_calendar: 02:00
 
+  gitaly:
+    signing_key:
```
Comment (chr4) [inline pillar.example]: Mind to add two examples here? One with `ssh-ed25519 AAA...` and one with a dummy GPG key?

Comment (zadjadr) [reply pillar.example]: Added!

**Comment by chr4** [comment]
Awesome! Thanks for your contribution!

**Pull request merged**